crates/wit-component/tests/linking.rs

```diff
@@ -0,0 +1,270 @@
+use {
+    anyhow::{Context, Result},
+    std::{borrow::Cow, path::Path},
+    wasm_encoder::{CustomSection, Encode as _, Module, RawSection},
+    wasmparser::Parser,
+    wit_component::StringEncoding,
+    wit_parser::{Resolve, UnresolvedPackage},
+};
+
+const FOO: &str = r#"
+(module
```

Comment:
One comment I'd have on this is that I think it'd be best if the test suite here could be structured like the other wit-component tests where there's a folder-of-tests and each test is a set of files configuring what needs to be done. The test-runner is more-or-less "go use the library equivalents of `wasm-tools` CLI commands to execute tests".

I've opened https://github.com/bytecodealliance/wasm-tools/pull/1135 which may be able to assist with that, but I've found this style of tests is beneficial for reading tests, writing tests, and debugging tests. (e.g. basically making it such that it's possible to play with everything on the CLI rather than requiring custom Rust code to drive everything.

Reply:
I've merged #1135 into https://github.com/dicej/wasm-tools/pull/new/linking-wat and added linking support to the components.rs test runner.  When 1135 is merged, I 'll rebase this PR and pull that in.  Meanwhile, I'll continue adding tests.

BTW, is it okay if I keep the one end-to-end test in linking.rs in addition to all the WAT tests?  It would be nice to have at least one test that actually runs the component.

Reply:
I've finished adding tests to the `linking-wat` branch of my fork, so just waiting on 1135 to land at this point.